lms/services/youtube.py

```diff
@@ -1,21 +1,74 @@
+import json
+
+from lms.services import SerializableError
+from lms.services.http import HTTPService
+
+
+class VideoNotFound(SerializableError):
+    def __init__(self, video_id):
+        super().__init__(error_code="youtube_video_not_found", details={"video_id": video_id})
+
+
 class YoutubeService:
-    """An interface for dealing with Youtube API."""
+    """An interface for dealing with YouTube API."""
 
-    def __init__(self, enabled: bool):
+    def __init__(self, enabled: bool, api_url: str, api_key: str):
         """
-        Initialise the Youtube service.
+        Initialise the YouTube service.
 
-        :param enabled: Whether Youtube is enabled on this instance
+        :param enabled: Whether YouTube is enabled on this instance
+        :param api_url: YouTube API base URL
+        :param api_key: YouTube API key
         """
         self._enabled = enabled
+        self._api_url = api_url
+        self._api_key = api_key
+        self._http = HTTPService()
 
     @property
     def enabled(self) -> bool:
-        """Get whether this instance is configured for Youtube."""
+        """Get whether this instance is configured for YouTube."""
+
+        return bool(self._enabled and self._api_url and self._api_url)
+
+    def video_info(self, video_id: str) -> dict:
+        """
+        Fetch YouTube video information.
+
+        :param video_id: A YouTube video ID
+        :raise VideoNotFound: If the video cannot be found
+        """
+
+        resp = self._http.get(url=f"{self._api_url}/videos", params={
+            "id": video_id,
+            "key": self._api_key,
+            "part": "contentDetails,snippet",
+            "maxResults": "1",
+        })
 
-        return self._enabled
+        # TODO Be more strict on response schema
+        json_resp: dict = json.loads(resp.text)
```

Comment:
I reckon if you do resp.json directly you get some extra check (eg are the headers right for a json repsons).

If you want to go full schema definition then it will be something like:

```SchemaClass(response).parse()```

and you get a dict back

like we do for example in

lms/services/d2l_api/client.py
lms/services/jstor/_article_metadata.py

Reply:
> I reckon if you do resp.json directly...

Oh, is that possible? My IDE did not autocomplete that method 😅, but that's worth doing in this PR.

Reply:
ok, perhaps I'm blind, haha
![image](https://github.com/hypothesis/lms/assets/2719332/c36b6524-4305-48d3-9a3f-8be058dc9405)

Reply:
For now, I personally wouldn't bother with the schema, maybe a link to the docs for `/video` close to where we use it.